pkg/console/controllers/downloadsdeployment/controller.go

```diff
@@ -0,0 +1,139 @@
+	operatorclientv1 "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	operatorinformerv1 "github.com/openshift/client-go/operator/informers/externalversions/operator/v1"
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/console/status"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/console-operator/pkg/console/controllers/util"
+	deploymentsub "github.com/openshift/console-operator/pkg/console/subresource/deployment"
+)
+
+type DownloadsDeploymentSyncController struct {
+	operatorClient v1helpers.OperatorClient
+	// configs
+	operatorConfigClient       operatorclientv1.ConsoleInterface
+	infrastructureConfigClient configclientv1.InfrastructureInterface
+	// core kube
+	deploymentClient appsv1.DeploymentsGetter
+	// events
+	resourceSyncer resourcesynccontroller.ResourceSyncer
+}
+
+func NewDownloadsDeploymentSyncController(
+	// top level config
+	configClient configclientv1.ConfigV1Interface,
+	// clients
+	operatorClient v1helpers.OperatorClient,
+	operatorConfigClient operatorclientv1.ConsoleInterface,
+	// informer
+	configInformer configinformer.SharedInformerFactory,
+	operatorConfigInformer operatorinformerv1.ConsoleInformer,
+	// core kube
+	deploymentClient appsv1.DeploymentsGetter,
+	deploymentInformer appsinformersv1.DeploymentInformer,
+	// events
+	recorder events.Recorder,
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+) factory.Controller {
+	configV1Informers := configInformer.Config().V1()
+
+	ctrl := &DownloadsDeploymentSyncController{
+		// configs
+		operatorClient:             operatorClient,
+		operatorConfigClient:       operatorConfigClient,
+		infrastructureConfigClient: configClient.Infrastructures(),
+		// client
+		deploymentClient: deploymentClient,
+		// events
+		resourceSyncer: resourceSyncer,
+	}
+
+	configNameFilter := util.NamesFilter(api.ConfigResourceName)
+	downloadsNameFilter := util.NamesFilter(api.OpenShiftConsoleDownloadsDeploymentName)
+
+	return factory.New().
+		WithFilteredEventsInformers( // infrastructure configs
+			configNameFilter,
+			operatorConfigInformer.Informer(),
+			configV1Informers.Infrastructures().Informer(),
+		).WithFilteredEventsInformers( // downloads deployment
+		downloadsNameFilter,
+		deploymentInformer.Informer(),
+	).WithSync(ctrl.Sync).
+		ToController("ConsoleDownloadsDeploymentSyncController", recorder.WithComponentSuffix("console-downloads-deployment-controller"))
+}
+
+func (c *DownloadsDeploymentSyncController) Sync(ctx context.Context, controllerContext factory.SyncContext) error {
```

Comment:
is this copied from somewhere?

Reply:
well all the controllers have the similar `Sync()` method. not sure why you are asking?